electrumx/server/controller.py

```diff
@@ -863,3 +863,21 @@ def mempool_get_fee_histogram(self):
         branch = [hash_to_hex_str(hash) for hash in branch]
 
         return {"block_height": height, "merkle": branch, "pos": pos}
+
+    async def transaction_get_id_from_pos(self, height, tx_pos):
+        '''Return the txid given a block height and position
+        in the block.
+        '''
+        height = self.non_negative_integer(height)
+        tx_pos = self.non_negative_integer(tx_pos)
+
+        hex_hashes = await self.daemon_request('block_hex_hashes', height, 1)
+        block_hash = hex_hashes[0]
+        block = await self.daemon_request('deserialised_block', block_hash)
+        tx_hashes = block['tx']
+        try:
+            tx_hash = tx_hashes[tx_pos]
+        except IndexError:
+            raise RPCError(BAD_REQUEST, f'no tx at position {tx_pos} in '
+                           f'block {block_hash} at height {height:,d}')
+        return tx_hash
```

Comment:
60% of the logic here is common with transaction_get_merkle.  Can you break out a common async subroutine, perhaps tx_hashes_at_height?  It would validate height and do thhe daemon_request calls.  Finally also format tx_pos with ":,d" in the error message.